packages/@lwc/state/tsconfig.json

```diff
@@ -4,7 +4,7 @@
     "module": "nodeNext",
     "moduleResolution": "nodeNext",
     "esModuleInterop": false,
-    "target": "ES2021",
+    "target": "ES2023",
```

Comment:
Required to use non-registered symbols as WeakMap keys, more[ details here](https://github.com/microsoft/TypeScript/pull/54195)

Reply:
reverted to es2021, as per our discussion